### PR TITLE
Timetable Refractoring done

### PR DESCRIPTION
**Pull request by venkylm10**
completes #128 
@Hareesh-Nandigrama for some courses api reponse has wrong data, apart from that timetable displays correctly according to new reponse.

### CODE REVIEW

File: lib/models/timetable/course_model.dart [hunk 1]
```diff
@@ -30,6 +37,9 @@ class CourseModel implements Comparable<CourseModel> {
     venue = json['venue'];
     midsem = json['midsem'];
     endsem = json['endsem'];
+    midSemVenue = json['midSemVenue'];
+    endSemVenue = json['endSemVenue'];
```
Comment (DeltaProxima) [inline lib/models/timetable/course_model.dart:41]: the response fields are midsemVenue and endsemVenue where S is capital
correct these properly

File: lib/models/timetable/course_model.dart [hunk 2]
```diff
@@ -9,17 +9,24 @@ class CourseModel implements Comparable<CourseModel> {
   String? venue;
   String? midsem;
   String? endsem;
+  String? midSemVenue;
+  String? endSemVenue;
+  Map<String, dynamic>? timings;
```
Comment (DeltaProxima) [inline lib/models/timetable/course_model.dart:14]: why are there 2 fields timings and timing

File: lib/models/timetable/course_model.dart [hunk 3]
```diff
@@ -30,6 +37,9 @@ class CourseModel implements Comparable<CourseModel> {
     venue = json['venue'];
     midsem = json['midsem'];
     endsem = json['endsem'];
+    midSemVenue = json['midSemVenue'];
+    endSemVenue = json['endSemVenue'];
+    timings = json['timings'];
```
Comment (DeltaProxima) [inline lib/models/timetable/course_model.dart:42]: You cannot map json['timimgs'] directly you need to appropriately convert it to a map<string,dynamic> and then map it

File: lib/widgets/timetable/exam_schedule_tile.dart [hunk 4]
```diff
@@ -100,26 +99,25 @@ class ExamTile extends StatelessWidget {
...
-                      //         ),
-                      //         Expanded(
-                      //           child: Text(
-                      //             course.venue!,
-                      //             style: MyFonts.w400.size(13).setColor(lBlue),
-                      //           ),
-                      //         )
-                      //       ],
-                      //     )
+                      if (course.venue != null && course.venue!.isNotEmpty)
```
Comment (DeltaProxima) [inline lib/widgets/timetable/exam_schedule_tile.dart:102]: It should be midsem or endsem venue appropriately

**Review by DeltaProxima — CHANGES_REQUESTED** [review]
@venkylm10 make all the required changes and test again

File: lib/models/timetable/course_model.dart (hunk 1 above)
Comment (venkylm10) [reply lib/models/timetable/course_model.dart]: done

File: lib/models/timetable/course_model.dart (hunk 2 above)
Comment (venkylm10) [reply lib/models/timetable/course_model.dart:14]: removed timing field

File: lib/models/timetable/course_model.dart (hunk 3 above)
Comment (venkylm10) [reply lib/models/timetable/course_model.dart]: fixed

File: lib/widgets/timetable/exam_schedule_tile.dart (hunk 4 above)
Comment (venkylm10) [reply lib/widgets/timetable/exam_schedule_tile.dart]: fixed

**Comment by venkylm10** [comment]
@DeltaProxima , Its complete now, can you verify once

**Comment by DeltaProxima** [comment]
@Hareesh-Nandigrama

**Review by DeltaProxima — APPROVED** [review]

**Pull request merged**